test/TinyRAM/Spec/EndToEndSpec (copy).hs

```diff
@@ -0,0 +1,509 @@
+
+cmpaGreaterTestCase :: Spec
+cmpaGreaterTestCase = 
+  before (handleCommand (CommandParse (AssemblyFilePath "examples/cmpaGreaterTest.s") objectFilePath)) $
+    it "The CMPA operation should result in setting the flag to 1." $ do
+      program <- readObjectFile objectFilePath
+      let answer = executeProgram (Params 16 16) (Just 1000) program (InputTape [1,2,3,4]) (InputTape [1,2,3])
+      answer `shouldBe` Right (1)
+  where 
+    objectFilePath = ObjectFilePath "examples/cmpaGreaterTest.o"
+
+cmpeGreaterTestCase :: Spec
+cmpeGreaterTestCase = 
+  before (handleCommand (CommandParse (AssemblyFilePath "examples/cmpeGreaterTest.s") objectFilePath)) $
+    it "The CMPE operation should result in setting the flag to 0." $ do
+      program <- readObjectFile objectFilePath
+      let answer = executeProgram (Params 16 16) (Just 1000) program (InputTape [1,2,3,4]) (InputTape [1,2,3])
+      answer `shouldBe` Right (0)
+  where 
+    objectFilePath = ObjectFilePath "examples/cmpeGreaterTest.o"
+
+cmpgeGreaterTestCase :: Spec
+cmpgeGreaterTestCase = 
+  before (handleCommand (CommandParse (AssemblyFilePath "examples/cmpgeGreaterTest.s") objectFilePath)) $
+    it "The CMPGE operation should result in setting the flag to 1." $ do
+      program <- readObjectFile objectFilePath
+      let answer = executeProgram (Params 16 16) (Just 1000) program (InputTape [1,2,3,4]) (InputTape [1,2,3])
+      answer `shouldBe` Right (1)
+  where 
+    objectFilePath = ObjectFilePath "examples/cmpgeGreaterTest.o"
+
+cmpgGreaterTestCase :: Spec
+cmpgGreaterTestCase = 
+  before (handleCommand (CommandParse (AssemblyFilePath "examples/cmpgGreaterTest.s") objectFilePath)) $
+    it "The CMPG operation should result in setting the flag to 1." $ do
+      program <- readObjectFile objectFilePath
+      let answer = executeProgram (Params 16 16) (Just 1000) program (InputTape [1,2,3,4]) (InputTape [1,2,3])
+      answer `shouldBe` Right (1)
+  where 
+    objectFilePath = ObjectFilePath "examples/cmpgGreaterTest.o"
+
+jmpTestExample :: Spec
+jmpTestExample = 
+  before (handleCommand (CommandParse (AssemblyFilePath "examples/jmpTestExample.s") objectFilePath)) $
+    it "The jmp test example should result in 0." $ do
+      program <- readObjectFile objectFilePath
+      let answer = executeProgram (Params 16 16) (Just 1000) program (InputTape [1,2,3,4]) (InputTape [1,2,3])
+      answer `shouldBe` Right (0)
+  where 
+    objectFilePath = ObjectFilePath "examples/jmpTestExample.o"
+
+jmpTestExampleNonTerm :: Spec
+jmpTestExampleNonTerm = 
+  before (handleCommand (CommandParse (AssemblyFilePath "examples/jmpTestExampleNonTerm.s") objectFilePath)) $
+    it "The jmp test example should result in non termination." $ do
+      program <- readObjectFile objectFilePath
+      let answer = executeProgram (Params 16 16) (Just 1000) program (InputTape [1,2,3,4]) (InputTape [1,2,3])
+      answer `shouldBe` Right (0)
```

Comment:
If the program does not terminate `executeProgram` should return Left instead of Right. Please investigate this.